python/lsst/pipe/base/argumentParser.py

```diff
@@ -29,57 +29,16 @@
     "InputOnlyArgumentParser",
 ]
 
-import abc
 import argparse
-import collections
-import fnmatch
-import itertools
-import logging
-import os
-import re
-import shlex
-import shutil
-import sys
-import textwrap
 
-import lsst.daf.persistence as dafPersist
-import lsst.log as lsstLog
-import lsst.pex.config as pexConfig
-import lsst.pex.config.history
-import lsst.utils
-import lsst.utils.logging
-
-DEFAULT_INPUT_NAME = "PIPE_INPUT_ROOT"
-DEFAULT_CALIB_NAME = "PIPE_CALIB_ROOT"
-DEFAULT_OUTPUT_NAME = "PIPE_OUTPUT_ROOT"
-
-
-def _fixPath(defName, path):
-    """Apply environment variable as default root, if present, and abspath.
-
-    Parameters
-    ----------
-    defName : `str`
-        Name of environment variable containing default root path;
-        if the environment variable does not exist
-        then the path is relative to the current working directory
-    path : `str`
-        Path relative to default root path.
-
-    Returns
-    -------
-    abspath : `str`
-        Path that has been expanded, or `None` if the environment variable
-        does not exist and path is `None`.
-    """
-    defRoot = os.environ.get(defName)
-    if defRoot is None:
-        if path is None:
-            return None
-        return os.path.abspath(path)
-    return os.path.abspath(os.path.join(defRoot, path or ""))
+from deprecated.sphinx import deprecated
 
 
+@deprecated(
+    reason="Gen2 DataIdContainer is no longer supported. This functionality has been disabled.",
+    version="v23.0",
```

Comment:
In all of these, should the version listed be `v23.0` (current) or `v24.0` (next), thus promising that this routine will be in v24 but once that is released can be deleted? Reading the docs is ... unclear ... but that's how I read it: https://developer.lsst.io/stack/deprecating-interfaces.html#python-deprecation

Reply:
We already promised that Gen2 was deprecated and could be removed any time as part of the v23 release. My statement is that CmdLineTask has been deprecated since v23 but now actually tells you it has been deprecated since then. I don't really see any point in keeping this code around for another two release cycles where it doesn't actually do anything and will break if something uses it.  The "version" here is the version the code was formally deprecated. I have not included in the statements when we will actually delete the stubs but I was assuming I could do it as soon as nothing in lsst_distrib uses CmdLineTask.

Reply:
I talked it over with @ktlim and he thinks that saying "since v23" is fine here.